cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/MetricDeleteQuery.java

```diff
@@ -37,7 +38,7 @@ public MetricDeleteQuery(long startTs, long endTs, String metricName,
     this.startTs = startTs;
     this.endTs = endTs;
     this.metricName = metricName;
-    this.sliceByTagValues = ImmutableMap.copyOf(sliceByTagValues);
+    this.sliceByTagValues = Maps.newHashMap(sliceByTagValues);
```

Comment:
when you want to delete all apps in a namespace, you would specify , /v3/metrics/delete?context=namespace.somespace.app.* , the app tag will map to null value, and ImmutableMap doesn't allow null in key or value, that's why you need this change.